docs/source/github_workflow.rst

```diff
@@ -1,37 +1,91 @@
-Release Workflow
-==================
+Git and Release Workflow
+=========================
 
-When preparing a new release, developers need to be aware of the existence of multiple processes.
+When preparing a new release, developers must take care to correctly follow the release process.
 
-Branches
+Git Branches
 --------------
 
-* develop
-* master
-* release
+* **master**: This is the stable branch. Release tags are always on this branch and the latest release is always the head of this branch.
+* **develop**: This is the main development branch and can be ahead of the master branch.
+  One should never directly merge and push to develop but perform a pull request on GitHub.
+  The PR can only be merged if approved by another developer and all automatic tests pass.
+* **wip_xx**: Feature branches for feature number xx start with wip_xx (one can add a short name to the branch name).
+  They always branch off develop, and as explained above, are merged to develop via GitHub pull requests.
+* **release/vx.y.z**: These are release branches (for version vx.y.z). They always branch off from develop.
+  Once the release process is complete (see below), the release branch is merged to both master and develop via GitHub pull requests.
 
-.. Add schema + explanation ?
+GitHub Workflows
+-----------------
 
+This project uses a number of GitHub workflows to automate various CI/CD tasks.
+The table bellow gives an overview of which workflows are triggered by what events.
 
-Docker build
---------------
-TODO
+.. list-table:: Main GitHub Workflows Summary
+    :widths: 2 1 1 1 1 1 1
+    :header-rows: 1
+    :stub-columns: 1
 
-Helm
---------------
-TODO
-.. * 
+    * - Worflow / Trigger
+      - PR to develop
+      - PR to master
+      - Push to develop
+      - Push to release/**
+      - Push to master
+      - GitHub release
+    * - Tests and Linters
+      - Yes
+      - Yes
+      - No
+      - No
+      - No
+      - No
+    * - Docker build and push
+      - No
+      - No
+      - Yes (tags git sha)
```

Comment:
what does tags git and sha mean ?

Reply:
The tag of the dockerfile is the git commit sha. I'll try to make it clearer without it taking too much space.